types/proposal.go

```diff
@@ -41,11 +41,22 @@ func (p *Proposal) String() string {
 }
 
 func (p *Proposal) WriteSignBytes(chainID string, w io.Writer, n *int, err *error) {
-	wire.WriteTo([]byte(Fmt(`{"chain_id":"%s"`, chainID)), w, n, err)
-	wire.WriteTo([]byte(`,"proposal":{"block_parts_header":`), w, n, err)
-	p.BlockPartsHeader.WriteSignBytes(w, n, err)
-	wire.WriteTo([]byte(Fmt(`,"height":%v,"pol_block_id":`, p.Height)), w, n, err)
-	p.POLBlockID.WriteSignBytes(w, n, err)
-	wire.WriteTo([]byte(Fmt(`,"pol_round":%v`, p.POLRound)), w, n, err)
-	wire.WriteTo([]byte(Fmt(`,"round":%v}}`, p.Round)), w, n, err)
+
+	wire.WriteJSON(
+		struct {
+			ChainID          string        `json:"chain_id"`
+			BlockPartsHeader PartSetHeader `json:"block_parts_header"`
```

Comment:
all this should be encapsulated in "proposal". Can we even do nested anonymous structs?